features/netsocket/TCPSocket.cpp

```diff
@@ -60,16 +63,15 @@ nsapi_error_t TCPSocket::connect(const SocketAddress &address)
         } else {
             blocking_connect_in_progress = true;
 
-            int32_t count;
+            uint32_t flag;
 
             // Release lock before blocking so other threads
             // accessing this object aren't blocked
             _lock.unlock();
-            count = _write_sem.wait(_timeout);
+            flag = _event_flag.wait(WRITE_FLAG, osFlagsWaitAny, _timeout);
             _lock.lock();
-
-            if (count < 1) {
-                // Semaphore wait timed out so break out and return
+            if (0 == (flag & WRITE_FLAG)) {
```

Comment:
One of the surprising annoyances of the new osEventFlags API, is that wait actually returns error codes on timeout:
https://www.keil.com/pack/doc/CMSIS/RTOS2/html/group__flags__error__codes.html

These conditions should probably be something like:
``` cpp
if ((flag & WRITE_FLAG) != WRITE_FLAG) {
```